sirepo_bluesky/tests/test_stateless_compute.py

```diff
@@ -0,0 +1,197 @@
+        "wallThickness": "80.e-06",
+    }
+
+    diff = list(dictdiffer.diff(browser_request, expected_response))
+    assert diff
+    print(diff)
+
+    classes, objects = create_classes(srw_chx_simulation.data, connection=srw_chx_simulation)
+
+    crl1 = objects["crl1"]
+
+    print(crl1.summary())  # noqa
+    pprint.pprint(crl1.read())  # noqa
+
+    RE(bps.mv(crl1.tipRadius, 150))
+
+    actual_response = srw_chx_simulation.compute_crl_characteristics(crl1.id._sirepo_dict)
+
+    assert actual_response.pop("state") == "completed"
+
+    pprint.pprint(actual_response)
+    newresponse1 = _remove_dict_strings(actual_response)
+    newresponse2 = _remove_dict_strings(expected_response)
+    pprint.pprint(newresponse1)
+    assert newresponse1.any()
+
+    assert np.allclose(newresponse1, newresponse2)
+
+
+@vcr.use_cassette(f"{cassette_location}/test_crl_characteristics.yml")
+def test_stateless_compute_crl_characteristics_advanced(srw_chx_simulation, tmp_path):
+    classes, objects = create_classes(srw_chx_simulation.data, connection=srw_chx_simulation)
+
+    _generate_test_crl_file(tmp_path / "test_crl_characteristics.json", objects["crl1"], srw_chx_simulation)
+
+    with open(tmp_path / "test_crl_characteristics.json", "r") as fp:
+        combined_dict = json.load(fp)
+    assert combined_dict
+
+    for key in combined_dict["request"]:
+        actual_response = srw_chx_simulation.compute_crl_characteristics(combined_dict["request"][key])
+        expected_response = combined_dict["response"][key]
+        actual_response.pop("state")
+        assert np.allclose(_remove_dict_strings(actual_response), _remove_dict_strings(expected_response))
+        print(f"{key} passed")
+
+
+@vcr.use_cassette(f"{cassette_location}/test_crystal_characteristics.yml")
+def test_stateless_compute_crystal(srw_tes_simulation, tmp_path):
+    classes, objects = create_classes(srw_tes_simulation.data, connection=srw_tes_simulation)
+
+    _generate_test_crystal_file(
+        tmp_path / "test_compute_crystal.json", objects["mono_crystal1"], srw_tes_simulation
+    )
+
+    with open(tmp_path / "test_compute_crystal.json", "r") as fp:
+        combined_dict = json.load(fp)
+    assert combined_dict
+
+    for key in combined_dict["request"]:
+        actual_init = srw_tes_simulation.compute_crystal_init(combined_dict["request"][key])
+        expected_init = combined_dict["init"][key]
+        actual_init.pop("state")
+        assert np.allclose(_remove_dict_strings(actual_init), _remove_dict_strings(expected_init))
+
+        actual_orientation = srw_tes_simulation.compute_crystal_orientation(combined_dict["request"][key])
+        expected_orientation = combined_dict["orientation"][key]
+        actual_orientation.pop("state")
+        assert np.allclose(_remove_dict_strings(actual_orientation), _remove_dict_strings(expected_orientation))
+        print(f"{key} passed")
+
+
+def _remove_dict_strings(dict):
+    newarr = np.array([], dtype=np.float64)
```

Comment:
Do we need a numpy array, or can it just be a list (`[]`) or `collections.deque()`?